compiler/rustc_span/src/source_map.rs

```diff
@@ -462,6 +462,17 @@ impl SourceMap {
         )
     }
 
+    pub fn span_to_location_info(&self, sp: Span) -> (Symbol, usize, usize, usize, usize) {
+        if self.files.borrow().source_files.is_empty() || sp.is_dummy() {
+            return (Symbol::intern("no-location"), 0, 0, 0, 0);
+        }
+
+        let lo = self.lookup_char_pos(sp.lo());
+        let hi = self.lookup_char_pos(sp.hi());
+        let sf_sym = lo.file.name.display(FileNameDisplayPreference::Local).to_string();
+        (Symbol::intern(&sf_sym), lo.line, lo.col.to_usize() + 1, hi.line, hi.col.to_usize() + 1)
```

Comment:
Maybe return `Lrc<SourceFile>` instead of a symbol, so the caller can decide what to do with it

Reply:
done

Reply:
> Please rebase the merge commits away and squash your fixup commits (or just make one big commit, that's also ok).

done

> Do we have documentation somewhere for the existing json output of the test suite?

none, the json was implemented without an RFC and through a tracking issue #49359. i will be creating the combined RFC for all of this and we'll have the documentation there. till then the agreement is to leave it under the unstable_opts flag.